docs/CodingGuidelines.md

```diff
@@ -0,0 +1,184 @@
+   *GOOD ASSERTION EXAMPLE*:  
+   
+   `expect(result).toEqual(expect.objectContaining({ error: null })`
+
+   *BAD ASSERTION EXAMPLE*:  
+   
+   every test asserting on every property over and over whether or not they are relevant to the purpose of the test
+
+1. Tests names should be descriptive and from a users' perspective:  
+1. 
+   *GOOD NAME EXAMPLE*: “it creates new user when new user form is submitted”  
+   
+   *BAD NAME EXAMPLE*: “button dispatches onClick event”
+
+1. Tests should, at minimum, cover all core feature use-cases and identifiable edge-cases or boundary conditions.
+
+1. Code must be written in a consistent style, utilizing the shared eslint and stylelint configurations https://github.com/awslabs/iot-app-kit/blob/main/.eslintrc.js https://github.com/awslabs/iot-app-kit/blob/main/.stylelintrc
+1. Pull requests to feature branches must contain only a single, well formatted commit that adheres to https://www.conventionalcommits.org/en/v1.0.0/
+1. Pull requests to main must have correct change logs which properly document migrations, feature changes, and most importantly breaking change. [example commit](https://github.com/awslabs/iot-app-kit/commit/9e09d6229dc5d27309766db4717d5a250e91bdd7)
+1. Pull requests to main from a feature branch should maintain the git history as contained in the feature branch
+1. Pull requests to main must have the correct versioning updates, following https://semver.org/
+1. Pull request descriptions should include a reasonably detailed summary of changes and highlight any architectural decisions introduced.
+1. Reviewers are expected to enforce a high standard on:
+    1. Consistent code to rest of codebase
+    1. High quality tests that are not painful to maintain
+    1. Architectural and big picture aspects of how the change fits in
+1. It is ok to build out work incrementally within a feature branch, but not in main.
+
+    For example, it is OK to have a PR merged into a feature branch with missing test cases and add those in the next PR. This helps allow people to get code in more frequently rather than having long lived local branched
+
+    When a pull request to main is made, there will be no “rolling fixes/improvements in next PR”.
+12. Avoid long-lived local feature-branches. Make pull requests incrementally, even if not everything is perfect. Explicitly state in the PR overview what aspects are in progress to help reviewers effectively understand what parts to provide feedback on.
+13. Customer facing APIs should not require ENUMS. Strings with strict typescript types are preferable and easier for customers to utilize.
+
+    *GOOD EXAMPLE*:
+     ```
+    <iot-line-chart size=“SM” ... /> 
+    ```
+
+    *BAD EXAMPLE*:
+    ```  
+    <iot-line-chart size={Size.SM} ... /> 
+    ```  
+
+14. Make the code consistent with the existing code base
+15. Avoid mutations when possible. Prefer immutability
+    *GOOD EXAMPLE*:  
```

Comment:
Need a line break here, it's inlining on the previous line

Reply:
fixed